### ISSUE TITLE
Changing default monitored metric for ReduceLRonPlateau from "val_loss" to "loss"

### ISSUE THREAD

**Issue by alanlegoallec** [tensorflow/tensorflow#50477]
- TensorFlow version: v2 (not relevant, feature change request is documentation-based)
- Willing to contribute it: Yes, it should be a one line change

Based on the documentation and my experience with tf, the default monitored metric for the callback ReduceLRonPlateau is "val_loss".
I suggest it should be changed to "loss" (the training loss).
My understanding of the rational for reducing the learning rate is that a learning rate that is too high prevents the model from converging to the nearby local optima. While I understand that the end goal is to minimize the validation loss and that monitoring this quantity is tempting, I do not understand the rational for monitoring it to reduce the learning rate.

Specifically, I consider 4 situations as a 2x2 matrix. The training loss is either decreasing, or it is not. Similarly, the validation loss is either decreasing, or it is not.

(1) If both the training and the validation losses are decreasing, then it does not matter which one is monitored (and the learning rate should not be changed).

(2) Similarly, if neither the training loss nor the validation loss is decreasing, then it does not matter which one is monitored (and the learning rate should be decreased).

(3) Now, if the training loss is decreasing, but the validation loss is not, then there are two explanations: 
(A) we are likely overfitting and reducing the learning rate is going to make the problem worse. 
(B) The validation loss is actually going down on average, but it is too noisy to be captured by the model. In that case too, reducing the learning rate is premature. 
/!\ However this is what the current default metric choice does. /!\ 
At this point the model should probably be early stopped anyway, but if we decide to keep training it, then we should not make the problem worse (maybe we should even increase the learning rate to help the model escape its local minima, but this is not the point of this post). 

(4) If the training loss is not decreasing but the validation loss is, then I would agree that the current default metric could be better. However I fail to see how this could happen? The only two scenarios I could imagine are:
(A) if the training loss is jumping from one local minima to another, with similar train losses but some local minima leading to better generalization. In that case I agree that reducing the learning rate would lead to premature converging in a local minima, but again this scenario seems unlikely to me.
(B) More likely, there is a higher standard error on the validation metric, and no actual underlying improvement of the generalization of the model. The learning rate should be reduced to improve the training loss, but the current default does not allow it.

I would agree that this is in the grey area, but so far the coworkers with whom I shared my thoughts ended up agreeing that monitoring the training loss makes more sense and is more intuitive. Some figures do not even mention the validation set in the context of learning rate reduction:
![ReduceLROnPlateau](https://user-images.githubusercontent.com/16949397/123551658-c6e97700-d740-11eb-9eb4-d999a33d6791.png)
Source: https://towardsdatascience.com/the-subtle-art-of-fixing-and-modifying-learning-rate-f1e22b537303

I personally do not mind changing this parameter for my projects, and I anticipate that the difference in performance will be small, but I did not find any issue regarding this question, so I wanted to at least raise it.

Thank you.

**Comment by fchollet** [comment]
The intent of the callback is to modulate the learning rate when the model's generalization is starting to degrade (overfitting). The set of cases where the *training loss* will enter a plateau is limited: this is unlikely to occur in very high dimensional spaces. Typically the training loss will always keep decreasing even if the model is in a regime where it would benefit from a lower learning rate.

But if your intent is to modulate the learning rate based on the training loss, simply passing `monitor='loss'` will do it.
